test/general-tests/amarna/check.ts

```diff
@@ -0,0 +1,3 @@
+import { hardhatStarknetTest } from "../../utils/cli-functions";
+
+hardhatStarknetTest("--no-compile test/amarna.test.ts".split(" "));
```

Comment:
Instead of executing `npx hardhat test` on a script from the example repo, why not just have all that logic inside this `check.ts` script? I.e. it seems to me that this could be tested without the separate example-repo PR (https://github.com/Shard-Labs/starknet-hardhat-example/pull/91). That's what I was aiming at with this comment: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/295#pullrequestreview-1263803398

Reply:
Aah true.
🤦‍♂️

Reply:
Aah, now I remember... It needs Cairo files in the root of the project tho. So it seemed to make sense to put in the example plug-in.

But we could move it here too.

Reply:
The tests are executed from the root of the project, so you can count on the source files to be present.

Check this test case for example: https://github.com/Shard-Labs/starknet-hardhat-plugin/tree/master/test/general-tests/expect-error-on-compile

It needs its own cairo file so it first copies it into the example project (in which everything is executed).

Reply:
Aah yeah, it does. Fantastic!

Reply:
Hi @FabijanC,

This is a great suggestion! Now we can call `npx hardhat amarna` instead of using `AmarnaDocker` internally.
Which also tests if the class is properly bound to the task. Brill!